### PR TITLE
feat: add 4 grids

### PR DESCRIPTION
**Pull request by daybrush**

### CODE REVIEW

**Review by mixed — APPROVED** [review]
LGTM

File: src/GroupManager.ts [hunk 1]
```diff
@@ -8,24 +9,29 @@ import { InfiniteGridGroup, InfiniteGridItemInfo } from "./types";
 import { categorize, flat, makeKey, splitGridOptions } from "./utils";
 
 export interface GroupManagerOptions extends GridOptions {
-  gridConstructor: GridFunction;
+  gridConstructor: GridFunction | null;
```
Comment (WoodNeck) [inline src/GroupManager.ts:12]: Why would this have to receive `null`? Looks like it is required in all places.

Comment (daybrush) [reply src/GroupManager.ts:12]: right. It is required, but I had to put null in defaultOptions.

Comment (WoodNeck) [reply src/GroupManager.ts:12]: Well... I think it's okay if you're fine with that but maybe that's kinda excessive.
Using `Omit` could be one solution I guess.

**Review by WoodNeck — APPROVED** [review]

**Comment by daybrush** [comment]
The type is determined by the type of defaultOptions and propertyTypes, and the getter is determined by defaultOptions. What about typecasting?

**Pull request merged**